static/src/javascripts/projects/common/modules/identity/upsell/account-creation/AccountCreationForm.js

```diff
@@ -20,41 +19,43 @@ class AccountCreationForm extends Component {
             errors: [],
         });
 
-        reqwest({
-            url: '/password/guest',
-            method: 'post',
-            data: {
-                csrfToken: this.props.csrfToken,
-                token: this.props.accountToken,
-                password: this.state.password,
-            },
-            success: () => {
-                ophan.record({
-                    component: 'set-password',
-                    value: 'set-password',
+        fetch(
+            '/password/guest',
+            {
+                method: 'POST',
+                body: JSON.stringify({
+                    csrfToken: this.props.csrfToken,
+                    token: this.props.accountToken,
+                    password: this.state.password,
+                })
+            }
+        )
+        .then( () => {
```

Comment:
Not sure how reqwest works but `fetch` will return  a 4** response as success. Correctly so (because the network call was successful) but if the other lib behaves differently this might cause a change in behaviour.

Reply:
If it's useful, in DCR we access the `ok` field to [check for errors](https://github.com/guardian/dotcom-rendering/blob/c2dc86aef6b425035d9d669438f2ae5b32660581/src/web/lib/api.tsx#L29)

Reply:
That’s a good point. I think this is a good candidate for `fetch-json`, which will throw an error if the status code isn’t `ok`.